### ISSUE TITLE
`--tmp-logdir` not working

### ISSUE THREAD

**Issue by runiq** [easybuilders/easybuild-framework#2935]
When setting `--tmp-logdir`, I was expecting the build logs etc. to be placed into the directory. However, nothing of the sort happens: The logs are still put into `--tmpdir`. A simple example:

```bash
mkdir $HOME/tmplogs
eb -l --debug --tmp-logdir=$HOME/tmplogs zlib-1.2.11.eb
```

Nothing is ever in `$HOME/tmplogs` during or after the build.

Bootstrapped EasyBuild 3.9.2 on CentOS 7.6.

**Comment by ocaisa** [comment]
Looking into this, it seems like the value of the log file location has been set before the configuration variable has been taken into account.

However, I'm not sure that this option does what you want, the `--tmp-logdir` option will only set a specific (temporary) destination for the easybuild logs, which will be cleaned up if a build is successful (after the log has been copied to the installation directory). You're command line is explicitly telling eb to log to the terminal (the `-l` option) so even if it worked this directory would still be empty, and if a build was successful it would also be empty without the `-l` option. What behaviour do you want exactly?

**Comment by runiq** [comment]
Sorry, the `-l` was just for illustration purposes in this case—I'm normally running `eb` without it.

> the --tmp-logdir option will only set a specific (temporary) destination for the easybuild logs, which will be cleaned up if a build is successful (after the log has been copied to the installation directory).

That's exactly what I'm expecting, actually :)

**Comment by ocaisa** [comment]
@boegel I'm stuck on this one. I believe the solution is to change https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/tools/build_log.py#L201 to
```
fd, logfile = tempfile.mkstemp(suffix='.log', prefix='easybuild-', dir=get_build_log_path())
```
but this leads to an circular import

**Comment by ocaisa** [comment]
Dancing around the circular import also does not help, I'll open a PR with what I have to see if I am missing something stupid

**Comment by ocaisa** [comment]
See #2937

**Comment by boegel** [comment]
Fix merged via #2937, will be included in next EasyBuild release (most likely 3.9.3), thanks a lot for reporting this issue @runiq!
